Vic2ToHoI4/Source/Configuration.cpp

```diff
@@ -2,28 +2,25 @@
 #include "Log.h"
 #include "OSCompatibilityLayer.h"
 #include "ParserHelpers.h"
+#include "StringUtils.h"
 #include <fstream>
 #include <vector>
 
 
 
-std::unique_ptr<Configuration> ConfigurationDetails::importDetails(std::istream& theStream)
+Configuration::Factory::Factory()
 {
-	std::string HoI4Path;
-	std::string Vic2Path;
-	std::vector<std::string> Vic2Mods;
-	auto forceMultiplier = 0.0;
-	auto manpowerFactor = 0.0;
-	auto industrialShapeFactor = 0.0;
-	auto icFactor = 0.0;
-	auto ideologiesOptions = ideologyOptions::keep_major;
-	std::vector<std::string> specifiedIdeologies;
-	auto debug = false;
-	auto removeCores = true;
-	auto createFactions = true;
-	auto version = HoI4::Version();
-
-	registerKeyword("HoI4directory", [&HoI4Path](const std::string& unused, std::istream& theStream) {
+	registerKeyword("SaveGame", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::singleString filenameString(theStream);
+		inputFile = filenameString.getString();
+		const auto length = inputFile.find_last_of('.');
+		if ((length == std::string::npos) || (".v2" != inputFile.substr(length, inputFile.length())))
```

Comment:
substr takes for second argument length of the substr, not end location, but in case length is longer than the string it'll clip at the end, so that's why this works. Should be simply 3.

Reply:
Yep, that's intentional. I don't want extensions like .v2xyz to pass the check.